### PR TITLE
Basic solidity contract.

### PR DESCRIPTION
**Pull request by ArnaudBrousseau**
This is a basic solidity contract. Still more things to do (tests!) but this should enable us to test integration with the frontend (@amhunt is working on it)

### CODE REVIEW

File: contracts/Blockbin.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    // TODO: do we need other fields?
+    struct Cube {
+        bytes32 hash;
+        address owner;
+        bytes data;
+    }
+
+    // Main Storage structure for Cubes
+    mapping (bytes32 => Cube) allBlockbins;
```
Comment (JFickel) [inline contracts/Blockbin.sol]: should be cubes

File: contracts/Blockbin.sol [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        delete allBlockbins[hash];
+        return true;
+    }
+
+    function forceEmpty (bytes32 hash) onlyAdmin returns (bool success) {
+        delete allBlockbins[hash];
+        return true;
+    }
+
+    function dumpCube(bytes data) returns (bool success) {
```
Comment (JFickel) [inline contracts/Blockbin.sol]: bug! what if multiple transactions occur in the same block?

Comment (ArnaudBrousseau) [reply contracts/Blockbin.sol]: Yeah that's a tough one. In solidity there's not good way to get the transaction nonce: http://solidity.readthedocs.io/en/develop/units-and-global-variables.html#block-and-transaction-properties

We could do `sha3(data)` to approach something unique but then 2 people wouldn't be able to dump the same content in the same transaction.

Comment (ArnaudBrousseau) [reply contracts/Blockbin.sol]: After talking, we decided to let this go and generate this hash on the client-side. This also lets users choose arbitrary hashes ("named" cubes).

**Comment by ArnaudBrousseau** [comment]
~~@JFickel not sure why this PR is not reflecting the changes I made on the branch but here they are: https://github.com/ArnaudBrousseau/blockbin/commits/basic_solidity_contract (see the latest commits)~~ EDIT: this PR is now reflecting the commits I pushed. We're all good!

Tell me if that looks good to you! I added a couple of basic tests to make sure things work properly (dump/read a cube)

**Comment by JFickel** [comment]
the web3 library is included with the truffle testing environment. fixed the tests a little

**Pull request merged**